lab_1_classify_by_unigrams/main.py

```diff
@@ -23,35 +31,65 @@ def tokenize(text: str) -> list[str] | None:
 
 
 def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
-    """
-    Calculate frequencies of given tokens.
+    if isinstance(tokens, list):
+        if isinstance(tokens[0], str):
+            frequency = {}
+            tokens_number = len(tokens)
+            for letter in tokens:
+                if isinstance(letter, str):
+                    letter_number = tokens.count(letter)
+                    letter_index = letter_number/tokens_number
+                    letter_dict = {letter: letter_index}
+                    frequency.update(letter_dict)
+            """
+            Calculate frequencies of given tokens.
 
-    Args:
-        tokens (list[str] | None): A list of tokens
+            Args:
+                tokens (list[str] | None): A list of tokens
 
-    Returns:
-        dict[str, float] | None: A dictionary with frequencies
+            Returns:
+                dict[str, float] | None: A dictionary with frequencies
 
-    In case of corrupt input arguments, None is returned
-    """
+            In case of corrupt input arguments, None is returned
+            """
+
+            return frequency
 
 
 def create_language_profile(language: str, text: str) -> dict[str, str | dict[str, float]] | None:
-    """
-    Create a language profile.
+    if isinstance(text, str) and isinstance(language, str):
+        freq_dict = calculate_frequencies(tokenize(text))
+        language_profile = {"name": language,
+                            "freq": freq_dict
+                            }
+        """
+        Create a language profile.
 
-    Args:
-        language (str): A language
-        text (str): A text
+        Args:
+            language (str): A language
+            text (str): A text
 
-    Returns:
-        dict[str, str | dict[str, float]] | None: A dictionary with two keys – name, freq
+        Returns:
+            dict[str, str | dict[str, float]] | None: A dictionary with two keys – name, freq
 
-    In case of corrupt input arguments, None is returned
-    """
+        In case of corrupt input arguments, None is returned
+        """
+        return language_profile
```

Comment:
the return value must be in the function body

Reply:
done